test/cpp/interop/client.cc

```diff
@@ -92,17 +93,68 @@ DEFINE_int32(soak_iterations, 1000,
 DEFINE_int32(iteration_interval, 10,
              "The interval in seconds between rpcs. This is used by "
              "long_connection test");
+DEFINE_string(additional_metadata, "",
+              "Additional metadata to send in each request, as a "
+              "semicolon-separated list of key:value pairs.");
 
 using grpc::testing::CreateChannelForTestCase;
 using grpc::testing::GetServiceAccountJsonKey;
 using grpc::testing::UpdateActions;
 
+namespace {
+
+// Parse the contents of FLAGS_additional_metadata into a map. Allow
+// alphanumeric characters and dashes in keys, and any character but semicolons
+// in values.
+std::multimap<grpc::string, grpc::string> ParseAdditionalMetadataFlag(
+    const grpc::string& flag) {
+  std::multimap<grpc::string, grpc::string> additional_metadata;
+
+  // Key in group 1; value in group 2.
+  std::regex re("([-a-zA-Z0-9]+):([^;]*);?");
```

Comment:
I think we cannot use regex because we still support gcc4.8, where regex is not supported properly according to: https://stackoverflow.com/questions/8561850/compile-stdregex-iterator-with-gcc

Reply:
Are there any string-processing utilities that we can use that'd help here?

Reply:
We mostly just hand-rolled stuff...
For a similar function, see https://github.com/grpc/grpc/blob/0fc01a302f03c313d33492439366821cb66e0eb6/test/cpp/util/grpc_tool.cc#L125

Reply:
Here's a hand-rolled version. I didn't try to make this one consistent with the one in grpc_tool.cc, but let me know if it'd be better to do that.